an atom may not have `@` attribute

This should be prohibited:

```
[@] > foo /number
```

Syntactically, it'a s valid EO program, but it must be prohibited to use `@` in atoms. Let's issue a `critical` defect in this case.

@yegor256 currently with `0.49.2` parser version, I getting this:

```text
[ERROR] Failed to parse '/Users/aliaksei/e/parsed/foo.eo:2': [2:1] error: no viable alternative at input:
[@] > foo /number
 ^
EO sources parsed in .eoc
```

XMIR:

```xml
<program xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         dob="2024-12-20T13:59:52"
         ms="31"
         name="foo"
         revision="b7b0283"
         source="/Users/aliaksei/e/parsed/foo.eo"
         time="2024-12-26T17:51:55.602703Z"
         version="0.49.2"
         xsi:noNamespaceSchemaLocation="https://www.eolang.org/xsd/XMIR-0.49.2.xsd">
   <listing># Foo.
[@] &gt; foo /number
</listing>
   <objects/>
   <errors>
      <error check="eo-parser" line="2" severity="critical">[2:1] error: no viable alternative at input:
[@] &gt; foo /number
 ^</error>
   </errors>
   <sheets>
      <sheet>stars-to-tuples</sheet>
      <sheet>wrap-method-calls</sheet>
      <sheet>const-to-dataized</sheet>
   </sheets>
</program>
```

As far I understand, `φ` in atoms already handled at the parser level, and we can close this issue, correct?

--- a/src/main/resources/org/eolang/lints/critical/atom-with-phi.xsl
+++ b/src/main/resources/org/eolang/lints/critical/atom-with-phi.xsl
@@ -27,7 +27,7 @@ SOFTWARE.
   <xsl:output encoding="UTF-8" method="xml"/>
   <xsl:template match="/">
     <defects>
-      <xsl:apply-templates select="//o[@atom][.//o[@name='@']]" mode="with-phi"/>
+      <xsl:apply-templates select="//o[@atom and o[@name='@']]" mode="with-phi"/>
     </defects>
   </xsl:template>
   <xsl:template match="o" mode="with-phi">
@@ -36,7 +36,7 @@ SOFTWARE.
         <xsl:value-of select="eo:lineno(@line)"/>
       </xsl:attribute>
       <xsl:attribute name="severity">critical</xsl:attribute>
-      <xsl:text>Atoms must not have '@' attribute "</xsl:text>
+      <xsl:text>Atom must not have '@' attribute "</xsl:text>
     </defect>
   </xsl:template>
 </xsl:stylesheet>

--- a/src/main/resources/org/eolang/lints/critical/atom-with-phi.xsl
+++ b/src/main/resources/org/eolang/lints/critical/atom-with-phi.xsl
@@ -36,7 +36,7 @@ SOFTWARE.
         <xsl:value-of select="eo:lineno(@line)"/>
       </xsl:attribute>
       <xsl:attribute name="severity">critical</xsl:attribute>
-      <xsl:text>Atoms must not have void '@' attribute "</xsl:text>
+      <xsl:text>Atoms must not have void '@' attribute</xsl:text>
     </defect>
   </xsl:template>
 </xsl:stylesheet>

--- a/src/main/resources/org/eolang/lints/critical/atom-with-phi.xsl
+++ b/src/main/resources/org/eolang/lints/critical/atom-with-phi.xsl
@@ -1,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+The MIT License (MIT)
+
+Copyright (c) 2016-2024 Objectionary.com
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included
+in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-->
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:eo="https://www.eolang.org" id="atom-with-phi" version="2.0">
+  <xsl:import href="/org/eolang/funcs/lineno.xsl"/>
+  <xsl:output encoding="UTF-8" method="xml"/>
+  <xsl:template match="/">
+    <defects>
+      <xsl:apply-templates select="//o[@atom][.//o[@name='@']]" mode="with-phi"/>
+    </defects>
+  </xsl:template>
+  <xsl:template match="o" mode="with-phi">
+    <defect>
+      <xsl:attribute name="line">
+        <xsl:value-of select="eo:lineno(@line)"/>
+      </xsl:attribute>
+      <xsl:attribute name="severity">critical</xsl:attribute>
+      <xsl:text>Atoms must not have '@' attribute "</xsl:text>
+    </defect>
+  </xsl:template>
+</xsl:stylesheet>

--- a/src/main/resources/org/eolang/lints/critical/atom-with-phi.xsl
+++ b/src/main/resources/org/eolang/lints/critical/atom-with-phi.xsl
@@ -36,7 +36,7 @@ SOFTWARE.
         <xsl:value-of select="eo:lineno(@line)"/>
       </xsl:attribute>
       <xsl:attribute name="severity">critical</xsl:attribute>
-      <xsl:text>Atom must not have '@' attribute "</xsl:text>
+      <xsl:text>Atoms must not have void '@' attribute "</xsl:text>
     </defect>
   </xsl:template>
 </xsl:stylesheet>

--- a/src/main/resources/org/eolang/motives/critical/atom-with-phi.md
+++ b/src/main/resources/org/eolang/motives/critical/atom-with-phi.md
@@ -1,0 +1,17 @@
+# Atom With `@`
+
+Atoms must not have `@` attribute.
+
+Incorrect:
+
+```eo
+# Foo.
+[@] > foo /number
+```
+
+Correct:
+
+```eo
+# Foo.
+[] > foo /number
+```

--- a/src/test/resources/org/eolang/lints/packs/atom-with-phi/allows-atoms-without-phi.yaml
+++ b/src/test/resources/org/eolang/lints/packs/atom-with-phi/allows-atoms-without-phi.yaml
@@ -1,0 +1,29 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2016-2024 Objectionary.com
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+---
+sheets:
+  - /org/eolang/lints/critical/atom-with-phi.xsl
+asserts:
+  - /defects[count(defect[@severity='critical'])=0]
+input: |
+  # No comments.
+  [] > foo /number

--- a/src/test/resources/org/eolang/lints/packs/atom-with-phi/catches-atom-with-phi.yaml
+++ b/src/test/resources/org/eolang/lints/packs/atom-with-phi/catches-atom-with-phi.yaml
@@ -1,0 +1,30 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2016-2024 Objectionary.com
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+---
+sheets:
+  - /org/eolang/lints/critical/atom-with-phi.xsl
+asserts:
+  - /defects[count(defect[@severity='critical'])=1]
+  - /defects/defect[@line='2']
+input: |
+  # No comments.
+  [@] > foo /number

